sys/od/od.c

```diff
@@ -27,12 +28,28 @@ void od_hex_dump(const void *data, size_t data_len, uint8_t width)
         width = OD_WIDTH_DEFAULT;
     }
 
+    uint8_t str_pos = 0;
     for (unsigned int i = 0; i < data_len; i++) {
-        print_str("  ");
+        print_str(" ");
         print_byte_hex(((uint8_t *)data)[i]);
 
         if ((((i + 1) % width) == 0) || i == (data_len - 1)) {
-            puts("");
+            /* fill in whitespace for incomplete hex lines */
+            for (unsigned j = i; ((j + 1) % width) != 0; ++j) {
+                print_str("   ");
+            }
+            print_str("  ");
+            for(unsigned k = 0; k < (i % width) + 1; k++){
+                if(isprint(((char *)data)[str_pos+k])){
+                    putchar(((char *)data)[str_pos+k]);
+                }
+                else{
+                    print_str(".");
```

Comment:
I know @cgundogan asked for `print_str()`, but why use `print_str()` here, but `putchar()` above?